### PR TITLE
fix(agents): add `agent.yaml` translation support for new Fabric 0.2.0 fields

### PR DESCRIPTION
**Pull request by mmogallapalli**
## Summary

Extends the Platform-owned `nemo-agents-spec-v1` `agent.yaml` schema and Fabric translator to cover the three Fabric configuration paths requested for MCP policy, OpenTelemetry export, and deployment-scoped MCP bearer authentication.

After this change, an `agent.yaml` can carry per-server `allowed_tools` and `blocked_tools`, HTTP `custom_headers` containing secret-safe environment-variable references, and Relay OpenTelemetry configuration into the generated `FabricConfig`.

## Related Issue

None.

## Changes

- Add `allowed_tools` and `blocked_tools` to each Platform `McpServerConfig` and pass both fields through to Fabric's normalized MCP server configuration. An explicit empty allowlist is preserved rather than treated as an omitted value.
- Preserve Fabric's capability gate for MCP tool filtering. Fabric 0.2 includes the normalized fields, but none of its bundled Claude, Codex, Deep Agents, or Hermes adapters declares `mcp.tool_filters`; those adapters therefore reject filtered configurations during planning instead of silently ignoring the policy. The translated fields are usable by custom and future adapters that declare and enforce the capability.
- Add per-server `custom_headers` so HTTP MCP clients can use deployment-scoped bearer authentication such as `Authorization: Bearer ${MCP_ACCESS_TOKEN}`.
- Keep bearer token values out of `agent.yaml` and the staged MCP configuration. Existing environment resolution collects the referenced secret for process-environment injection while preserving the `${MCP_ACCESS_TOKEN}` reference in `custom_headers`. This does not add per-user token propagation.
- Add `telemetry.opentelemetry` and translate it to Relay's OpenTelemetry export configuration alongside ATIF and ATOF.
- Default a missing OpenTelemetry endpoint `service_name` to the agent name while preserving explicitly configured service names and all other endpoint options.
- Add schema, translation, and environment-resolution tests for MCP filters, explicit empty allowlists, bearer header references, OpenTelemetry endpoints, transport and header settings, resource attributes, and service-name behavior.

## Type of Change

- [x] Code change (feature, bug fix, or refactor)
- [ ] Code change with documentation updates
- [ ] Documentation only
- [ ] Contributor tooling or automation
- [ ] CI, build, or test infrastructure

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [ ] Documentation updated for user-visible behavior
- [x] Documentation not applicable — justification: this PR adds schema and translation parity; product documentation can be updated when a bundled adapter supports MCP tool filtering.

## Verification

- [x] Pull request title follows the repository's Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [ ] `uv run pre-commit run -a` passes, or any blocked checks are identified below
- [x] Targeted tests pass, or tests are marked not applicable above
- [x] No secrets, API keys, or credentials are included

Targeted validation:

- `.venv/bin/pytest plugins/nemo-agents/tests/unit/test_agent_config.py plugins/nemo-agents/tests/unit/test_fabric_translator.py plugins/nemo-agents/tests/unit/test_environment_resolution.py -q` — passed, 56 tests.
- Focused Ruff linting and formatting checks passed for all five changed Python files.
- Focused `ty` checks passed for the production schema and its configuration-model tests. Broader touched test modules retain unrelated pre-existing type diagnostics.
- Manual Fabric 0.2 planning accepted the translated OpenTelemetry and bearer-header configurations.
- Manual Fabric 0.2 planning rejected MCP tool-filter configurations for each bundled adapter with the expected missing-capability error, confirming the policies are not silently ignored.
- `git diff --check` — passed.

## Summary by CodeRabbit

* **New Features**
  * Added OpenTelemetry configuration support, including endpoint metadata, authentication mappings, resource attributes, and automatic service naming.
  * Added custom headers for MCP servers.
  * Added optional MCP tool allowlists and blocklists for more precise tool exposure control.
  * Preserved explicitly empty allowlists and existing server headers during configuration merging.

### CODE REVIEW

File: plugins/nemo-agents/src/nemo_agents_plugin/fabric/translator.py [hunk 1]
```diff
@@ -207,9 +207,31 @@ def _relay_observability_config(config: AgentConfig, model: ModelConfig) -> dict
...
+def _relay_opentelemetry_config(opentelemetry_config: dict[str, Any], agent_name: str) -> dict[str, Any]:
+    opentelemetry = dict(opentelemetry_config)
+    endpoints = opentelemetry.get("endpoints")
+    if not isinstance(endpoints, list):
+        return opentelemetry
+
+    enriched_endpoints: list[Any] = []
+    for endpoint_config in endpoints:
+        if isinstance(endpoint_config, dict):
+            endpoint_config = dict(endpoint_config)
```
Comment (mckornfield) [inline plugins/nemo-agents/src/nemo_agents_plugin/fabric/translator.py:228]: why this cast if it's this type? to satisfy `ty`?

**Review by mckornfield — APPROVED** [review]

File: plugins/nemo-agents/src/nemo_agents_plugin/fabric/translator.py (hunk 1 above)
Comment (mmogallapalli) [reply plugins/nemo-agents/src/nemo_agents_plugin/fabric/translator.py:228]: not for `ty`, it’s copying the nested dict so adding the default `service_name` doesn’t mutate the input config directly
